library/swift_disk.py

```diff
@@ -29,23 +38,39 @@
 def main():
     module = AnsibleModule(
         argument_spec  = dict(
-            dev       = dict(required=True),
+            dev       = dict(required=False, default=None),
+            partition_path      = dict(required=False, default=None)
         ),
     )
 
     dev = module.params.get('dev')
-    dev_path = "/dev/%s" % dev
-    part_path = "%s1" % dev_path
-
-    if not os.path.exists(dev_path):
+    part_path = module.params.get('partition_path')
+    
+    if (dev is None and part_path is None) or \
+          (dev is not None and part_path is not None) :
+        module.fail_json(msg="only one of the dev or partition_path must be set")
```

Comment:
There is a convenience method to express this.  See https://groups.google.com/forum/#!topic/ansible-devel/ny8YSlCWVSQ